flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/AbstractRocksDBState.java

```diff
@@ -154,6 +159,29 @@ public void setCurrentNamespace(N namespace) {
 		return backend.db.get(columnFamily, tmpKeySerializationView.getCopyOfBuffer());
 	}
 
+	public byte[] migrateSerializedValue(
+			byte[] serializedOldValue,
+			TypeSerializer<V> priorSerializer,
+			TypeSerializer<V> newSerializer) throws StateMigrationException {
+
+		try {
+			V value = priorSerializer.deserialize(new DataInputViewStreamWrapper(new ByteArrayInputStream(serializedOldValue)));
+
+			ByteArrayOutputStream baos = new ByteArrayOutputStream();
+			DataOutputViewStreamWrapper out = new DataOutputViewStreamWrapper(baos);
```

Comment:
Better use `DataInputDeserializer` / `DataOutputSerializer` instead of wrapped byte-array streams. They are quiet a bit faster.

Reply:
This method is called in a loop. You should avoid creating streams and other temporary objects on each invocation. Instead, let the calling loop pass them in and reuse the same instances for the whole process. If we had a `StateValueTransformer` instance, they could also hold them as members.